### PR TITLE
Jvenstad/instances with separate change

### PR DESCRIPTION
**Pull request by jonmv**
@mpolden please review.

### CODE REVIEW

**Comment by jonmv** [comment]
I'll add some more tests for various things, and perhaps discover some errors in the process, but the main things are done here.

**Review by mpolden — APPROVED** [review]
Great!

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java [hunk 1]
```diff
@@ -90,34 +90,42 @@ public DeploymentStatus(Application application, Map<JobId, JobStatus> allJobs,
...
+    /** Filterable job status lists for each instance of this application. */
     public Map<ApplicationId, JobList> instanceJobs() {
         return allJobs.asList().stream()
                       .collect(groupingBy(job -> job.id().application(),
                                           collectingAndThen(toUnmodifiableList(), JobList::from)));
     }
 
-    /** Returns the set of jobs that need to run for the application's current change to be considered complete. */
+    /**
+     * The set of jobs that need to run for the application's current change to be considered complete,
```
[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java]:
```suggestion
     * The set of jobs that need to run for the changes of each instance of the application to be considered complete,
```

[the remark this answers is not in the document]
Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/deployment/DeploymentStatus.java]: Yeah, I should probably look over the javadoc.

File: controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java [hunk 2]
```diff
@@ -699,24 +699,27 @@ private void toSlime(Cursor object, Application application, HttpRequest request
...
+        DeploymentStatus status = controller.jobController().deploymentStatus(application);
         application.latestVersion().ifPresent(version -> toSlime(version, object.setObject("latestVersion")));
 
         application.projectId().ifPresent(id -> object.setLong("projectId", id));
 
-        // Currently deploying change
-        if ( ! application.change().isEmpty())
-            toSlime(object.setObject("deploying"), application.change());
+        // TODO jonmv: Remove this when users are updated.
+        application.instances().values().stream().findFirst().ifPresent(instance -> {
```
Comment (jonmv) [inline controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: Ah, this should prefer production instances.

Comment (jonmv) [reply controller-server/src/main/java/com/yahoo/vespa/hosted/controller/restapi/application/ApplicationApiHandler.java]: Well, I don’t think it’s used for anyrhing now.9

**Pull request merged**